main/tests/test_nllb_cased.rs

```diff
@@ -0,0 +1,260 @@
+mod test_utils;
+
+use rust_tokenizers::tokenizer::{
+    MultiThreadedTokenizer, NLLBTokenizer, Tokenizer, TruncationStrategy,
+};
+use rust_tokenizers::{Offset, TokenizedInput};
+use test_utils::download_file_to_cache;
+
+#[test]
+fn test_nllb_tokenization() -> anyhow::Result<()> {
+    let vocab_path = download_file_to_cache(
+        "https://huggingface.co/facebook/nllb-200-distilled-600M/resolve/main/tokenizer.json",
+        "nllb_tokenizer.json",
+    )
+    .unwrap();
+
+    let merges_path = download_file_to_cache(
+        "https://huggingface.co/facebook/nllb-200-distilled-600M/resolve/main/sentencepiece.bpe.model",
+        "nllb_sentencepiece.model",
+    )
+    .unwrap();
+
+    let special_path = download_file_to_cache(
+        "https://huggingface.co/facebook/nllb-200-distilled-600M/raw/main/special_tokens_map.json",
+        "nllb_special.json",
+    )
+    .unwrap();
+
+    let nllb_tokenizer = NLLBTokenizer::from_files(vocab_path, merges_path, special_path)?;
+
+    let original_strings = [
+        "nld_Latn …",
+        "eng_Latn This is a sample sentence to be tokénized",
+        "eng_Latn Wondering how this will get tokenized 🤔 ?",
+        "fra_Latn İs th!s 𩸽 Ϻ Šœ Ugljšić dấu nặng",
+        "hin_Deva   İs th!s    𩸽 Ϻ Šœ   Ugljšić  dấu nặng     ",
+        "lit_Latn � İs th!s �� 𩸽 Ϻ Šœ   Ugljšić  dấu nặng     ",
+    ];
+
+    let expected_results = [
+        TokenizedInput {
+            token_ids: vec![256127, 622, 2],
```

Comment:
It looks like the EOS token (id 265047) is consistently missing when compared to the output from the Python tokenizer. The behaviour of the tokenizer should be identical to the Python version, the integration tests help capture such discrepancies

Reply:
Hm, looks like you're right. But I don't know why it's behave like this and this code works fine for my `rust-bert` translator.

Do you have any ideas why it's skipping EOS?

Reply:
I just checked and it is not skipping EOS (EOS is actually the token id `2`). The NLLB tokenizer has an extended suffix, as seen in https://github.com/huggingface/transformers/blob/2e35bac4e73558d334ea5bbf96a1116f7c0d7fb3/src/transformers/models/nllb/tokenization_nllb.py#L203 defined by [`eos`, `cur_lang_code`]. The `cur_lang_code` deaults to `eng_Latn`, and does not seem to be updated in most cases. I suggest the following:
- Add a `src_lang` field to the `NLLBTokenizer`, defaulting to `eng_Latn` at creation
- Add a `set_src_lang` method taking `(&mut self, src_lang)` as parameters to update this source language code if needed (do not expect to be useful in the standard case)
- Update the `build_input_with_special_tokens` to add the token, and mask/offset data after the EOS token

Reply:
Fixed in https://github.com/npatsakula/rust-tokenizers/pull/5.